build.gradle.kts

```diff
@@ -1,105 +1,67 @@
-import org.ajoberstar.grgit.Grgit
-import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
-
 plugins {
-    java
-    jacoco
-    `maven-publish`
-    id("org.ajoberstar.grgit") version "4.1.1"
+    id("java-library")
+    id("maven-publish")
     id("com.github.johnrengelman.shadow") version "7.1.2"
 }
 
-logger.lifecycle("""
-*******************************************
- You are building LWCX!
- If you encounter trouble:
- 1) Try running 'build' in a separate Gradle run
- 2) Use gradlew and not gradle
- 3) If you still need help, ask on Discord! https://discord.gg/PHpuzZS
-
- Output files will be in build/libs
-*******************************************
-""")
-
-if (!project.hasProperty("gitCommitHash")) {
-    apply(plugin = "org.ajoberstar.grgit")
-    ext["gitCommitHash"] = try {
-        Grgit.open(mapOf("currentDir" to project.rootDir))?.head()?.abbreviatedId
-    } catch (e: Exception) {
-        logger.warn("Error getting commit hash", e)
-
-        "no.git.id"
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(8))
     }
+    withSourcesJar()
 }
 
 repositories {
     mavenCentral()
-
-    maven { url = uri("https://repo.papermc.io/repository/maven-public/") }
-    maven { url = uri("https://maven.enginehub.org/repo/") }
-    maven { url = uri("https://repo.glaremasters.me/repository/public/") }
-    maven { url = uri("https://jitpack.io") }
+    maven("https://hub.spigotmc.org/nexus/content/repositories/snapshots/")
+    maven("https://maven.enginehub.org/repo/")
+    maven("https://repo.glaremasters.me/repository/towny/")
+    maven("https://ci.ender.zone/plugin/repository/everything/")
+    maven("https://repo.glaremasters.me/repository/public/")
+    maven("https://jitpack.io")
 }
 
 configurations {
     compileClasspath.get().extendsFrom(create("shadeOnly"))
 }
```

Comment:
If you don't use the shadeOnly configuration, you can delete this.

Imo it's cleaner to use the shadeOnly configuration and specify the dependencies to relocate to minimize issues with dependencies changing package paths.

Reply:
Good catch, removed. I haven't really seen shadeOnly configuration used in any other projects, so mostly out of familiarity I'd prefer just using implementation and relocating that in shadow, provided there aren't any other reasons against doing so.